src/EvoSC.Common/Services/MapService.cs

```diff
@@ -93,6 +93,26 @@ public async Task<IEnumerable<IMap>> AddMapsAsync(List<MapStream> mapStreams)
     public async Task RemoveMapAsync(long mapId)
     {
         await _mapRepository.RemoveMapAsync(mapId);
+        var map = await this.GetMapByIdAsync(mapId);
+
+        if (map == null) return;
+
+        var filePath = Path.Combine(_config.Path.Maps, "EvoSC", map.FilePath);
+        if (!File.Exists(filePath))
+        {
+            _logger.LogWarning("Tried to delete map which doesn't exists on the filesystem: {filePath}", filePath);
+            return;
+        }
+
+        try
+        {
+            File.Delete(filePath);
```

Comment:
It would be great to be able to test the actual file deletion, but I can't think about a way to verify the file deletion was handled with the correct path. @snixtho What do you think?